.github/workflows/build.yml

```diff
@@ -31,11 +31,13 @@ jobs:
       - name: Install Python
         uses: actions/setup-python@v5
         with:
-          python-version: "3.11"
+          python-version: "3.12"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip
-          pip install ruff
+          python -m pip install uv
+          uv venv
+          source .venv/bin/activate
+          uv sync
```

Comment:
You can keep this in if you want, but it will also run automatically whenever you use a `uv` command.